docs/design/locale_fallback_and_negotiation.md

```diff
@@ -0,0 +1,427 @@
+b. ⇒ or return a list ordered by match score:
+[ `es-CL`, `es-419`, `es-AR`, … `es-VE`, ..., `es`, `es-ES` ]  
+Note how `es-CL` is first, `es-419` next, a “laundry list” of LatAm countries follow, and how `es` and `es-ES` are at the very end.
+* The panel locale would be set to `fr-CL` (or the longer list?), the formatting will use `es-CL`, and the strings will be `es-419` because the (vertical) fallback would take care of it for UI.
+And if the API returns a list (case b.) then we can use that exact list for vertical fallback.
+
+We can provide “syntactic sugar” APIs that can wrap all this and solve the problem of multiple lists of “available locales” (formatting data, UI, TTS, speech recognition, subtitles, you name it).
+
+But we should also allow access to the lower level methods that return an ordered matching list.
+
+This can become an even more interesting / challenging problem if we want even more sources of locale-aware services to match: UI strings, locale-aware formatting (ICU4X), TTS, Speech Recognition, available keyboards, maybe more. And they should all match as best as we can.
+
+We can “chain” intersections of lists like so:
+
+User desired ∩ ICU4X locales ∩ Available UI ⇒ [ `es-CL`, `es-419`, `es-AR`, `es` ]  
+[`es-CL`, `es-419`, `es-AR`, `es` ] ∩ Available TTS: [ `es`, `es-419` ] ⇒ [ `es-419`, `es-CL` ]  
+[ `es-CL`, `es-419`, `es-AR`, `es` ] ∩ Available Speech Recognition: [ `es` ] ⇒ [ `es` ]
+
+#### Expose an API / class to get locale suggestions
+
+Acknowledgement: this was proposed by Zibi.
+
+In many cases the developers have to implement language pickers where they have to suggest a few likely locales to put at the top of the list.
+
+This is often based on existing “hints” like country codes (for example from the physical location, and / or the SIM provider) and already selected locales.
+
+For example Android looks at the country of the SIM provider, the country of the radio tower connected (~location), the top locale already selected (current UI locale).
+
+So if I set my locale to French Belgium, I buy a SIM in US, and I am currently in Canada, the suggestions are:
+
+* Dutch (Belgium): because of the country of the current UI locale (BE)
+* English (Canada): because of the current location
+* English (US): because of the SIM provider
+* French (Canada): because of the current location
+* Spanish (US): because of the SIM provider
+
+Cherokee / Hawaiian / Lakota (all US) are excluded because Android is not localized into these locales. French Belgium is excluded because it is already selected (as UI locale)
+
+The locale excluding some locales or not might be controlled with options.
+
+This does not have to be in the Language Matcher proper, but a lot of the know-how and a lot of the data is common between such functionality and the matcher.
+
+So even if we choose to expose this as a separate class instead of methods in the matcher, I look at language matcher + suggestions as one single component.
```

Comment:
```suggestion
So even if we choose to expose this as a separate class instead of methods in the matcher, we should look at language matcher + suggestions as one single component.
```